src/parthenon_manager.cpp

```diff
@@ -379,6 +379,15 @@ void ParthenonManager::RestartPackages(Mesh &rm, RestartReader &resfile) {
     std::vector<std::size_t> counts, offsets;
     std::size_t count_on_rank =
         resfile.GetSwarmCounts(swarmname, myBlocks, counts, offsets);
+    // Compute total count and skip this swarm if total count is zero.
+    std::size_t total_count = count_on_rank;
+#ifdef MPI_PARALLEL
+    MPI_Allreduce(&total_count, &total_count, 1, MPI_UNSIGNED_LONG_LONG, MPI_SUM,
```

Comment:
Can we guarantee that `size_t` is always `MPI_UNSIGNED_LONG_LONG`?

Reply:
No we can't, but I don't know another solution, without introducing a lot of preprocessor nonsense... and `size_t=unsigned long long` seems pretty stable, if not standard guaranteed, on modern systems.

Reply:
I just noticed that is the the same as in `MPIPrefixSum` you mention below.
How about adding a static_assert kind of thing to (so not introduce all kinds of versions/types, just check that it matches as a cheap better-safe-than-sorry approach)?
I'm happy if you make the final call on this. I'm already approving as is.

Reply:
Yeah this is a good idea. I can add that.

Reply:
Ok this was slightly more complicated than I thought it would be, but I got there:
1. `unsigned long long int` and `unsigned long int` are typically identical on modern systems, but the type system still treats them as different. So instead of checking type identity, I check `sizeof`. It does however assume `size_t` will always be an unsigned integral type. I think this is safe to assume, but I added a static assert for it anyway.
2. Checking for `size_t` and `unsigned long long` caused `cpplint` to yell at me because these classic types are verboten. So I added an exception in the linter for `static_assert`.